src/box/lua/trigger.c

```diff
@@ -253,3 +253,146 @@ box_lua_trigger_init(struct lua_State *L)
 	luaL_register_type(L, event_trigger_iterator_typename,
 			   trigger_iterator_methods);
 }
+
+/** Old API compatibility. */
+
+/**
+ * Checks positional arguments for luaT_event_reset_trigger.
+ * Throws an error if the format is not suitable.
+ */
+static void
+luaT_event_reset_trigger_check_positional_input(struct lua_State *L, int bottom)
+{
+	assert(lua_checkstack(L, bottom + 2));
+	/* Push optional arguments. */
+	while (lua_gettop(L) < bottom + 2)
+		lua_pushnil(L);
+	/*
+	 * (nil, callable) is OK, deletes the trigger
+	 * (callable, nil), is OK, adds the trigger
+	 * (callable, callable) is OK, replaces the trigger
+	 * no arguments is OK, lists all trigger
+	 * anything else is error.
+	 */
+	bool ok = true;
+	/* Name must be a string if it is passed. */
+	ok = ok && (lua_isnil(L, bottom + 2) || luaL_isnull(L, bottom + 2) ||
+		    lua_type(L, bottom + 2) == LUA_TSTRING);
+	ok = ok && (lua_isnil(L, bottom + 1) || luaL_isnull(L, bottom + 1) ||
+		    luaL_iscallable(L, bottom + 1));
+	ok = ok && (lua_isnil(L, bottom) || luaL_isnull(L, bottom) ||
+		    luaL_iscallable(L, bottom));
```

Comment:
AFAIU the old API only supports plain functions. You don't need to extend it to support callback objects.

Reply:
Mentioned in commit message that the new helper support any callable object, linked the commit to the issue.